screens/UserTimeCards.js

```diff
@@ -0,0 +1,170 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { Field, reduxForm, FieldArray } from 'redux-form'
+import { ScrollView, Text, View, TouchableOpacity, Image, ActivityIndicator, FlatList, Modal } from 'react-native'
+import { DismissKeyboard } from '../components/DismissKeyboard'
+import BackBtn from '../components/BackBtn'
+import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome'
+import Icon from 'react-native-vector-icons/Ionicons'
+import { getTimeCards, getUserTimeCards, formatDate, getMonthName, formatDateObj } from '../actions'
+import images from '../assets/images'
+import styles from '../styles'
+import { LocaleContext } from '../locales/LocaleContext'
+import DropDown from '../components/DropDown'
+import UserTimeCardsFilterForm from './UserTimeCardsFilterForm'
+import { api, dispatchFetchRequest, errorAlert, warningMessage } from '../constants/Backend'
+
+class UserTimeCards extends React.Component {
+  static navigationOptions = {
+    header: null
+  }
+  static contextType = LocaleContext
+
+  state = {
+  	timecardId: null,  	
+  	filteredUsrTimeCards: []
+  }
+
+  handleFilter = (values) => {
+  	var month = values.month;
+  	var year = values.year;
+  	var username = this.props.navigation.state.params.name;
+
+  	if (!month || !year) {
+      warningMessage('Please Choose Both Year and Month')
+      return
+    }
+		
+     dispatchFetchRequest(api.timecard.getByMonthYrUsr(year, month, username), {
+        method: 'GET',
+        withCredentials: true,
+        credentials: 'include',
+        headers: {}
+      },
+      response => {
+        response.json().then(data => {
+          this.setState({filteredUsrTimeCards: data.timeCards})
+        })
+      }).then()
+  }
+
+  componentDidMount() {
+  	this.props.getUserTimeCards(this.props.navigation.state.params.name)
+  	var defaultVal={
+  		year: new Date().getFullYear(),
+  		month: getMonthName(new Date().getMonth() + 1),
+  		username: this.props.navigation.state.params.name
+  	}
+  	this.handleFilter(defaultVal)
+  }
+
+  render() {
+    const { t } = this.context
+    const { usertimeCards, haveData, haveError, loading, timeCard } = this.props
+    const { filteredUsrTimeCards } = this.state
+
+    Item = ({ timecard }) => {
+      return ( 	
+      	<TouchableOpacity
+					onPress={() => {
+          	this.props.navigation.navigate('UserTimeCardDetail',{
+          		timecardId: timecard.id
+          	})
+          }}
+      		>  
+        	<View style={[{paddingHorizontal: 8, marginBottom: 10}]}>
+          	<View style={[styles.flex_dir_row, styles.paddingTopBtn8]}>
+            	<View style={{flex: 6}}>
+              	<Text>
+              		{formatDateObj(timecard.clockIn)}&nbsp;&nbsp;{new Date(timecard.clockIn).getUTCHours() + ':' + new Date(timecard.clockIn).getUTCMinutes()} 
```

Comment:
formatDate(timecard.clockIn) returns  like this  "February 19,2020" . not hours and minutes show in fomatting.

Reply:
I think it would return time part of the date as well. I used it in clock in screen and order summary screen. did you try it? 

export const formatDate = date => {
  const dateMillis = Date.parse(date.slice(0, date.length - 5))
  const dateObj = new Date(dateMillis)
  return dateObj.toLocaleString('en-TW', {
    dateStyle: 'long',
    timeZone: 'Asia/Taipei'
  })
}

Reply:
I did it, but this function works well in iPhone, not work in android.